include/state_machine.h

```diff
@@ -28,11 +27,12 @@ class StateMachine
   bool activate();
 
 private:
-  void constructObstacleMenu(State* from, State* walk, State* single_step, State* obstacles);
-  void constructWalkMenu(State* from, State* prev_gait);
-  void constructStepMenu(State* from, State* prev_gait);
-  void constructSofaMenu(State* from, State* next_gait, State* walk, State* single_step);
-  void constructStairsMenu(State* from, State* next_gait, State* walk, State* single_step);
+  void constructObstacleMenu(State* from);
+  void constructWalkMenu(std::list<State>::iterator lastState);
+  void constructStepMenu(std::list<State>::iterator lastState);
```

Comment:
I think it would be useful to create a `using` for this iterator type.
```cpp
private:
  using state_iterator = std::list<State>::iterator
  ...
```
Then you can use it instead of the long type
```suggestion
  void constructWalkMenu(state_iterator lastState);
  void constructStepMenu(state_iterator lastState);
```
This creates less clutter, since you use it quite often, and if we ever want to change the type of the container, we only have to change one line.

Reply:
Nice, much better. I will implement this